cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/deploy/LocalManager.java

```diff
@@ -107,6 +111,7 @@ public LocalManager(CConfiguration configuration, PipelineFactory pipelineFactor
     pipeline.addLast(new DeployDatasetModulesStage(datasetFramework));
     pipeline.addLast(new CreateDatasetInstancesStage(datasetFramework));
     pipeline.addLast(new CreateStreamsStage(id, streamAdmin, exploreFacade, exploreEnabled));
+    pipeline.addLast(new CreateSchedulesStage(store, scheduler));
```

Comment:
Another thing that is tricky is that a schedule might fire right away... before the app and its programs (that is, the schedulable actions) are registered. In that case, the action would fail. That is why I feel that this must be the last stage of deployment.